main.tf

```diff
@@ -1,37 +1,44 @@
 resource "aws_lb" "default" {
-  count           = "${length(keys(var.access_logs)) == 0 ? 1 : 0}"
-  name            = "${var.name}"
-  internal        = "${var.internal}"
-  security_groups = ["${concat(var.security_groups, list(aws_security_group.default.id))}"]
-  subnets         = ["${data.aws_subnet_ids.selected.ids}"]
-  idle_timeout    = "${var.idle_timeout}"
+  count           = length(keys(var.access_logs)) == 0 ? 1 : 0
+  name            = var.name
+  internal        = var.internal
+  security_groups = concat(var.security_groups, [aws_security_group.default.id])
+  subnets         = data.aws_subnet_ids.selected.ids
+  idle_timeout    = var.idle_timeout
 
   access_logs {
     bucket  = "log-${data.aws_region.current.name}-${data.aws_caller_identity.current.account_id}"
     prefix  = "lb"
     enabled = "true"
   }
 
-  tags = "${var.tags}"
+  tags = var.tags
 
-  ip_address_type            = "${var.ip_address_type}"
-  enable_deletion_protection = "${var.enable_deletion_protection}"
-  enable_http2               = "${var.enable_http2}"
+  ip_address_type            = var.ip_address_type
+  enable_deletion_protection = var.enable_deletion_protection
+  enable_http2               = var.enable_http2
 }
 
 resource "aws_lb" "user" {
-  count           = "${length(keys(var.access_logs)) != 0 ? 1 : 0}"
-  name            = "${var.name}"
-  internal        = "${var.internal}"
-  security_groups = ["${concat(var.security_groups, list(aws_security_group.default.id))}"]
-  subnets         = ["${data.aws_subnet_ids.selected.ids}"]
-  idle_timeout    = "${var.idle_timeout}"
+  count           = length(keys(var.access_logs)) != 0 ? 1 : 0
+  name            = var.name
+  internal        = var.internal
+  security_groups = concat(var.security_groups, [aws_security_group.default.id])
+  subnets         = data.aws_subnet_ids.selected.ids
+  idle_timeout    = var.idle_timeout
 
-  access_logs = ["${var.access_logs}"]
+  dynamic "access_logs" {
```

Comment:
A dynamic block seems like overkill here. Will this work:

```
access_logs = var.access_logs
```

Reply:
I recommend you try testing the module by creating a load balancer without access logs.

Reply:
As far as I can tell, the behavior of the migrated module is one-for-one with the current behavior, and I have the tests to prove it.

I've filed an issue #6 for future work to streamline the lb code and make it more flexible. However it is likely that any refactoring we do in the future will be a breaking change due to the structural changes to the Terraform state (or at the very least will require using `terraform import`).